### ISSUE TITLE
U-net segmentation issue

### ISSUE THREAD

**Issue by llemgruber** [HenriquesLab/ZeroCostDL4Mic#5]
I’m trying to use the U-net part to segment EM images. Whenever I try to do the training, I always get a “blurry” result image, different from the predicted output - images attached.  
Most likely it is a problem with me (e.g. setting the wrong parameters). But unfortunately I’m a little lost in this.
I’m happy to share with you the folder, if that’s easier.
![28_predict](https://user-images.githubusercontent.com/62759899/77774510-1d9a6980-7043-11ea-98a7-38e1b3efe234.png)
![predicted_0](https://user-images.githubusercontent.com/62759899/77774528-23904a80-7043-11ea-8391-8f06b4673f92.png)

**Comment by lucpaul** [comment]
Hi, 
I just need a little bit more information. Are you using the data we provide with the notebooks or your own data? 
To make sure I understand, the upper image is the prediction that is shown in the notebook and the lower image is another result image that is shown? Do both of them appear like this in the notebook after playing the prediction?

**Comment by llemgruber** [comment]
Hiya
thanks for getting back to me quickly.
Yes, I was using the data set provided; the upper image is a predicted one in the notebook; the bottom one the result image after the training and still using the data set in the notebook.

**Comment by lucpaul** [comment]
Hey, no worries, I'm not sure what's going on, could you send a screenshot of where the blurry picture shows up?

**Comment by llemgruber** [comment]
Hi

here are some images with the parameters I used. The blurry image appears at the end, on step 5.1 "Generate Prediction from test dataset".
Thanks a lot for the help
<img width="844" alt="Screenshot 2020-03-27 at 16 34 57" src="https://user-images.githubusercontent.com/62759899/77778819-8258c280-7049-11ea-8cb4-81a9d6ab1307.png">
<img width="850" alt="Screenshot 2020-03-27 at 16 35 15" src="https://user-images.githubusercontent.com/62759899/77778841-8c7ac100-7049-11ea-95fb-7b5aeeceb612.png">
<img width="786" alt="Screenshot 2020-03-27 at 16 35 56" src="https://user-images.githubusercontent.com/62759899/77778852-91d80b80-7049-11ea-8508-ed80e9d6ec12.png">
<img width="777" alt="Screenshot 2020-03-27 at 16 36 38" src="https://user-images.githubusercontent.com/62759899/77778866-97cdec80-7049-11ea-97c3-88ae8eed04de.png">

**Comment by lucpaul** [comment]
Thank you, on first sight I don't really see anything wrong. 
I will just see if I can replicate this. In the meantime, maybe play around with some other parameter settings and see if that changes how the output looks. Maybe change the number of steps or epochs or even some of the other parameters. Also, make sure that there isn't some old model saved in your model folder that might give you this output.

**Comment by llemgruber** [comment]
I did changed some of the parameters, but the results were the same :/
But I'll give another try.
Thanks for the support

**Comment by lucpaul** [comment]
Ok, I just tested it, and I can't see what exactly is happening, my results look fine. One thing I noticed is that you are testing your prediction on 30 images although our actual test data should only have two images. So it looks like you are running the predictions on the same dataset as you trained the network with, is that right?
Generally, you shouldn't test your network on the same data that you trained it on. So try running the predictions on unseen input images. If you haven't downloaded the test set that is provided, just remove one image from the training set and run the prediction on that. How does your output look when you do that?

**Comment by lucpaul** [comment]
I tried it again with the training set and am still not quite sure. But we will put out a new release, probably today, so you should try it with the new version of the Unet notebook. There are a few changes in the new notebook so maybe the problem will disappear. If not, just let me know and I will have another look.

**Comment by llemgruber** [comment]
Thanks!

**Comment by llemgruber** [comment]
Hi. I tried the new notebook, and I keep having the same error - attached.

[error.txt](https://github.com/HenriquesLab/ZeroCostDL4Mic/files/4402265/error.txt)

**Comment by lucpaul** [comment]
Hi, I will need a little bit more information to help you with this error. Which cell is throwing the error? And what folders/values have you used for setting up training? By first glance, it might have to do with the 'visualisation after training' box. If you untick that and run again, how does it look? And how and where have you saved the training data?
If you like you can share a link to your notebook, maybe I can have a look directly. Maybe this will make it easier to find the error. I am sure we can get this to work.

**Comment by llemgruber** [comment]
Hi. It was on "Train the network" step.
Here is the link for the notebook
https://drive.google.com/open?id=1DTynwBKugFtlww-K15wgLyN0DPaaqwYQ

**Comment by llemgruber** [comment]
Really appreciate your help :)

**Comment by lucpaul** [comment]
I need to get access to your drive (have requested), or alternatively, you can just share the notebook as a link. No worries, I just want the notebooks to work smoothly, so thanks a lot for your comments. It will hopefully make them better.

**Comment by lucpaul** [comment]
Ok. I had a look at your notebook. I changed nothing except the path to the training folders and the training worked. So I suspect there might be an issue with the files in your folders. If you are using the data we provided, did you rename any of the files? Also, make sure there are no other files or folders in the training paths when you start training. All there should be in the source and target folder are the 2D .tif files, either the EM image (source) or the mask (target). 
Next, if you double click on the cells you can see and edit the code in colab. What might be causing an issue in cell 3, is the line starting with _base_directory_ near the top. If you have checked all of the above, try changing that line to just _base_directory = str(path.parent)_ and see if this can get rid of the error as this should get rid of the extra '/' in your path. It should not make a difference but might be worth trying.
But I think the problem might be in your file organisation (as the error is FileNotFound). So make sure that all the files are where they should be, that source and target files have the same names, are in the expected format and that you have no additional or missing files in each of the folders.

Let me know how you get on.

**Comment by llemgruber** [comment]
Hi. Thanks so much for your help with this issue.
After downloading the data, I get a folder "unet-master"; within it I get a folder "data"; where a folder "membrane" have the files "test-volume", "train-labels and "train-volume", plus the folders "test" and "train". What I did was create a folder (Volume) to put the "train-volume" file; and one folder (Label) for the "train-labels" file. I didn't change the file names. I run again but got the same problem, but I noticed that file names were not the same. So I changed to the same one "train" and it did run as it should be. But the training error figure does not has a validation curve - attached. I changed the batch size as well as epochs number and steps, all leading to the same issue. I'm really sorry for keeping bothering you with this.
<img width="893" alt="Screenshot 2020-03-30 at 17 30 38" src="https://user-images.githubusercontent.com/62759899/77940527-baae0a00-72b0-11ea-8ffc-f8425b50afbf.png">

**Comment by lucpaul** [comment]
Hi, thanks for continuing to try. I still think this issue might have to do with how your training data is organised. Please have a look at our wiki page on Unet. After looking at your comments we realised that we should make it clearer how to get and download the data, so we added some more explanations to our wiki on Unet. 
And then, the simplest way to get it to work for me was to just have the data in two folders, source and mask. And these folders should be in the same parent folder, e.g. "My Drive" (but any other location will work too).
So please have a look at the updated steps on our wiki and see if you can get it to work like that. And feel free to contact us here if you still have problems. It would be great if we can get this to work for you.

**Comment by llemgruber** [comment]
Hi. Thank you so, so much for your help. All working great now :)
I'm really sorry for all the messages and stupid questions.
:)

**Comment by lucpaul** [comment]
Very happy I could help you. The comments and questions were actually very helpful for us to improve our wiki and notebooks. So no need to apologize. Contact us if you run into other issues.
